### DISCUSSION TITLE
登录出现问题

### DISCUSSION THREAD

**Discussion by Agoines** [mamoe/mirai#1101] [Q&A]
在android设备上构建了一个Bot，如果是流量，Bot则无法登录，WiFi的话，Bot则可以正常登录

**Comment by mzdluo123** [comment]
检查你的dns设置有助于解决该问题，我的MiraiAndroid项目未遇到过此问题

**Reply by Agoines** [reply]
似乎是device没有自行配置的原因，然后Mirai Android 项目部分登录需要滑动解锁，但是如果采用登录设备的信息制作为device，使用后，似乎是不需要滑动解锁

**Reply by mzdluo123** [reply]
这个行为是正常现象，成功登录账户时所使用的设备信息可以在后续登录中无需验证

**Reply by Agoines** [reply]
> 检查你的dns设置有助于解决该问题，我的MiraiAndroid项目未遇到过此问题

似乎不是这个问题，采用的是
userBot = BotFactory.newBot(name, password) {
        BotConfiguration().apply { BotConfiguration.Default }
        protocol = BotConfiguration.MiraiProtocol.ANDROID_PAD // 切换协议
    }.alsoLogin()
在WiFi无设备锁，可以登录
在流量无设备锁情况下，无法登录，重复打印为以下log

W/Mirai: 未指定设备信息, 已使用随机设备信息. 请查看 BotConfiguration.deviceInfo 以获取更多信息.
    Device info isn't specified. Please refer to BotConfiguration.deviceInfo for more information
I/Net 3412436097: Connected to server 120.232.18.27:443
V/Net 3412436097: Send: wtlogin.login
V/Net 3412436097: Recv: LoginPacketResponse.Captcha.Picture
W/Mirai: 未指定设备信息, 已使用随机设备信息. 请查看 BotConfiguration.deviceInfo 以获取更多信息.
    Device info isn't specified. Please refer to BotConfiguration.deviceInfo for more information
I/Net 3412436097: Connected to server 183.3.235.162:14000
V/Net 3412436097: Send: wtlogin.login
V/Net 3412436097: Recv: LoginPacketResponse.Captcha.Slider

**Comment by Agoines** [comment]
目前尝试用 WIFI 登录成功一次，生成保存 device.json 以后登录调用生成的 device.json, 就可以随便登录，不会出现上文问题。

**Reply by Him188** [reply]
> V/Net 3412436097: Recv: LoginPacketResponse.Captcha.Slider

是因为用移动数据网络状态你的网络环境是不可信的, 服务器就要求了验证码

**Reply by Him188** [reply]
检查你的 `LoginSolver` 实现

**Reply by Agoines** [reply]
> 检查你的 `LoginSolver` 实现

感谢帮助，我去看看
